### PR TITLE
optimistic-transaction: remove reference to being "default"

### PR DESCRIPTION
**Pull request by nullnotnil**
# What is changed, added or deleted? (Required)

The documentation claimed optimistic transactions are the default. This is only true for v2.1.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: optimistic-transaction.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ aliases: ['/docs/dev/optimistic-transaction/','/docs/dev/reference/transactions/
 
 This document introduces the principles of TiDB's optimistic transaction model and related features.
 
-TiDB uses the optimistic transaction model by default. In TiDB's optimistic transaction model, for write-write conflicts, the two-phase commit begins only when the transaction is committed.
+In TiDB's optimistic transaction model, for write-write conflicts, the two-phase commit begins only when the transaction is committed.
```
Comment (jackysp) [inline optimistic-transaction.md:12]: The original Chinese is

> 对于写写冲突，只有事务提交时，才会通过两阶段提交检测是否存在写写冲突。

I think the English version is a bit confusing.

Comment (jackysp) [reply optimistic-transaction.md:12]:
```suggestion
Detecting the write-write conflict will only be triggered when committing the transaction via two-phase commit.
```

File: optimistic-transaction.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ aliases: ['/docs/dev/optimistic-transaction/','/docs/dev/reference/transactions/
 
 This document introduces the principles of TiDB's optimistic transaction model and related features.
 
-TiDB uses the optimistic transaction model by default. In TiDB's optimistic transaction model, for write-write conflicts, the two-phase commit begins only when the transaction is committed.
+In TiDB's optimistic transaction model, detecting the write-write conflicts only happens when committing the transaction via two-phase commit.
```
Comment (TomShawn) [inline optimistic-transaction.md:12]:
```suggestion
In TiDB's optimistic transaction model, write-write conflicts are detected only at the two-phase transactional commit.
```

**Pull request merged**